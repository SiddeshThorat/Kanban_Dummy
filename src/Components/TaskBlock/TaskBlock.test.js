import React from 'react';
import { shallow } from 'enzyme'
import TaskBlockComponent from'./TaskBlock.component';

describe("Testing TaskBlockComponent component",() => {

    const props = {
        category: "icebox",
        tasks: [
            {id:"1",title:"Dummy Task 1",category: "icebox"},
            {id:"2",title:"Dummy Task 2",category: "review"}
        ],
        onDragOver: jest.fn(),
        onDrop: jest.fn(),
        addTask:jest.fn()
    }

    it("Snapshot testing of TaskBlockComponent component",() => {
        expect(shallow(<TaskBlockComponent {...props}/>)).toMatchSnapshot();
    })
    it("Check if Header, containerForDraggable and InputField are rendered",() => {
        const component = shallow(<TaskBlockComponent {...props}/>)
        const header = component.find('Header')
        const containerForDraggable = component.find('.containerForDraggable');
        const inputField = component.find('InputField');

        expect(header.length).toEqual(1);
        expect(containerForDraggable.length).toEqual(1);
        expect(inputField.length).toEqual(1);

    })
})


