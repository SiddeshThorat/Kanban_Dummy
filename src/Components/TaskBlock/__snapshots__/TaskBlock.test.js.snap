// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TaskBlockComponent component Snapshot testing of TaskBlockComponent component 1`] = `
<div>
  <div
    className="containerIcebox"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <Header>
      Icebox
    </Header>
    <div
      className="containerForDraggable"
    />
    <InputField
      addTask={[Function]}
      category="icebox"
    />
  </div>
</div>
`;
